ibmcloud/README.md

```diff
@@ -119,9 +124,14 @@ $ terraform apply
 
 > **Tip:** You can check the status of provisioned Kubernetes node VM instances at [https://cloud.ibm.com/vpc-ext/compute/vs](https://cloud.ibm.com/vpc-ext/compute/vs).
 
+The SSH key installed on control-plane and worker nodes will be output at the end of the terraform configuration.
```

Comment:
Because of the way the IBM Cloud Terraform provider works with the `ibm_is_ssh_key` resource [reference here](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_ssh_key#argument-reference) the resource is forced to be recreated every time you run `terraform apply`.

If you provide the `ssh_pub_key` variable to let Terraform create the SSH key in VPC and run `terraform apply` again after running `terraform apply` once, the second `terraform apply` fails with an error like this:

```
ibm_is_ssh_key.created_ssh_key[0]: Destroying... [id=r038-06cafbfb-42b9-41ac-a718-57c55446ff53]
╷
│ Error: Error deleting SSH Key : SSH-key still in use by 2 instance(s). First instance ID: 02q7_16201105-5828-4673-ab88-2ad1625aa079
│ {
```

this is because the SSK key cannot be destroyed and replaced, as the Terraform provider wants to do, because the key is being used by the two Virtual Server instances.

I found a work around which involves one low-level Terraform state manipulation command, which nonetheless allows `terraform apply` to be re-ran, so I think we should document this work around for people who are providing the `ssh_pub_key` to this Terraform template. The work around is:

* Use `terraform state rm` to remove the created SSH key resource from Terraform's internal state:
```
terraform state rm ibm_is_ssh_key.created_ssh_key[0]
```
* Delete the `ssh_pub_key` variable from the `terraform.tfvars` file
* Run `terraform state list`, which should show the following output:

```
data.ibm_is_image.k8s_node
data.ibm_is_security_group.primary
data.ibm_is_ssh_key.ssh_key
data.ibm_is_subnet.primary
data.ibm_is_vpc.vpc
ibm_is_floating_ip.controlplane
ibm_is_floating_ip.worker
ibm_is_instance.controlplane
ibm_is_instance.worker
ibm_is_instance_template.k8s_node
local_file.inventory
null_resource.ansible
```

Then out can run `terraform apply` again. When I did that I got the output:

```
$ terraform apply -auto-approve
ibm_is_instance_template.k8s_node: Refreshing state... [id=02q7-c5ff2ead-15c1-4ace-818b-51b9776b4595]
ibm_is_instance.worker: Refreshing state... [id=02q7_0ccb2896-fb38-4c48-a4ec-0240c72ee61b]
ibm_is_instance.controlplane: Refreshing state... [id=02q7_f8746938-7e4e-43f7-a4b8-0723d7b5b8c6]
ibm_is_floating_ip.worker: Refreshing state... [id=r038-2c60be66-ee57-4fad-8eb9-3cfb353d8ec9]
ibm_is_floating_ip.controlplane: Refreshing state... [id=r038-e56b765c-b6b6-4a4c-875a-4f1cf11316fb]
local_file.inventory: Refreshing state... [id=8c4ea2f90ec12f5eff3012ed72ce72f506ca61c6]
null_resource.ansible: Refreshing state... [id=1766717801830837328]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

ssh_key_name = "toronto-key"
```

Reply:
Hi Matt, 

I've just tried running `terraform apply` multiple times with the ssh-pub-key set and it runs successfully without any workaround. Potentially something we can discuss separately.

Reply:
After Matt told me how he hit this I also managed to trigger this. My total case flow was:
- Don’t provide `ssh_pub_key` and use an existing key -> `terraform apply` which worked then `terraform destroy` to clean it up
- Provide  `ssh_pub_key` , but use the name of an existing key -> `terraform apply` which failed (correctly) with `[DEBUG] Create SSH Key Key with name already exists`
- Delete the existing key manually in IBM and re-run it and` terraform apply` which worked
- Re-run `terraform apply` without the destroy first, and then got:
```
ibm_is_ssh_key.created_ssh_key[0]: Destroying... [id=r022-727dba5f-bc61-42a9-9664-bb637444fdc3]
ibm_is_ssh_key.created_ssh_key[0]: Still destroying... [id=r022-727dba5f-bc61-42a9-9664-bb637444fdc3, 10s elapsed]
╷
│ Error: Error deleting SSH Key : SSH-key still in use by 2 instance(s). First instance ID: 02f7_0b218b6c-660e-484b-ba1d-785fd7fe1782
```
in case that helps?